File: src/components/ClipboardManager/utils/sensitive-data/detectors/detectEnvSecret.ts
import type { SensitiveMatch } from '../sensitiveData.types'

const ENV_SECRET_PATTERN = /^(?:(?:DATABASE_URL|DB_PASSWORD|API_KEY|API_SECRET|SECRET_KEY|AUTH_TOKEN|ACCESS_TOKEN|PRIVATE_KEY|ENCRYPTION_KEY|SIGNING_KEY|MASTER_KEY|APP_SECRET|JWT_SECRET|SESSION_SECRET|OAUTH_SECRET|CLIENT_SECRET)\s*=\s*.+)$/gim

export function detectEnvSecret(text: string): SensitiveMatch[] {
  const matches: SensitiveMatch[] = []
  ENV_SECRET_PATTERN.lastIndex = 0

  let match: RegExpExecArray | null
  while ((match = ENV_SECRET_PATTERN.exec(text)) !== null) {
    matches.push({
      type: 'env_secret',
      label: 'Env Secret',
      level: 'high',
      start: match.index,
      end: match.index + match[0].length,
    })
  }

  return matches
}

File: src-tauri/src/commands/clipboard/analysis/sensitive/detect_env_secret.rs
use super::super::utf16_offset::byte_to_utf16_offset;
use crate::types::SensitivityMatch;
use regex::Regex;
use std::sync::OnceLock;

/// Detects `.env`-style secret assignments. Faithful port of the frontend
/// `detectEnvSecret` sensitive-data detector. Matches are `high`.
pub fn detect_env_secret(text: &str) -> Vec<SensitivityMatch> {
    static RE: OnceLock<Regex> = OnceLock::new();
    let re = RE.get_or_init(|| {
        Regex::new(r"(?im)^(?:(?:DATABASE_URL|DB_PASSWORD|API_KEY|API_SECRET|SECRET_KEY|AUTH_TOKEN|ACCESS_TOKEN|PRIVATE_KEY|ENCRYPTION_KEY|SIGNING_KEY|MASTER_KEY|APP_SECRET|JWT_SECRET|SESSION_SECRET|OAUTH_SECRET|CLIENT_SECRET)\s*=\s*.+)$").expect("valid env-secret regex")
    });

    let mut matches = Vec::new();
    for m in re.find_iter(text) {
        matches.push(SensitivityMatch {
            kind: "env_secret".to_string(),
            label: "Env Secret".to_string(),
            level: "high".to_string(),
            start: byte_to_utf16_offset(text, m.start()),
            end: byte_to_utf16_offset(text, m.end()),
        });
    }
    matches
}
